TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/tripresult/UpdateTripForRealtime.kt

```diff
@@ -15,70 +16,85 @@ import java.util.concurrent.atomic.AtomicReference
 import javax.inject.Inject
 
 open class UpdateTripForRealtime @Inject internal constructor(
-        private val tripGroupRepository: TripGroupRepository,
-        private val errorLogger: ErrorLogger,
-        private val fetchingRealtimeStatusRepository: FetchingRealtimeStatusRepository,
-        private val tripUpdater: TripUpdater
+    private val tripGroupRepository: TripGroupRepository,
+    private val errorLogger: ErrorLogger,
+    private val fetchingRealtimeStatusRepository: FetchingRealtimeStatusRepository,
+    private val tripUpdater: TripUpdater
 ) {
-  private val subscriptions: CompositeDisposable = CompositeDisposable()
+    private val subscriptions: CompositeDisposable = CompositeDisposable()
 
-  fun start(getTripGroup: Observable<TripGroup>) {
-    Observable.interval(0, 10, TimeUnit.SECONDS, Schedulers.computation())
-        .withLatestFrom(getTripGroup, BiFunction<Long, TripGroup, TripGroup>{ _, tripGroup -> tripGroup } )
-        .filter { it.displayTrip!!.hasQuickBooking().not() }
-        .filter { it.displayTrip!!.updateURL  != null }
-        .flatMap { startAsync(it) }
-        .subscribe({
-          tripGroupRepository.updateTrip(it.second.uuid(), it.second.displayTrip!!.uuid(), it.first)
-              .subscribe({}, errorLogger::trackError )
-        }, errorLogger::logError)
-        .run {
-          subscriptions.add(this)
-        }
-  }
+    fun start(getTripGroup: Observable<TripGroup>) {
+        Observable.interval(0, 10, TimeUnit.SECONDS, Schedulers.computation())
+            .withLatestFrom(
+                getTripGroup,
+                BiFunction<Long, TripGroup, TripGroup> { _, tripGroup -> tripGroup })
+            .filter { it.displayTrip!!.hasQuickBooking().not() }
```

Comment:
### Code Review: Null Safety

The code currently relies on non-null assertions (`!!`) for certain properties, such as `it.displayTrip` and `it.displayTrip!!.updateURL`. To enhance the code's robustness and safety, consider the following improvements:

#### Use Safe Calls (?.)

Replace non-null assertions with safe calls to handle potential null values more gracefully. For example, instead of:

```kotlin
it.displayTrip!!.updateURL